File: RootTools2/src/main/java/com/rarnu/tools/neo/data/CompInfo.java
package com.rarnu.tools.neo.data;

import android.content.IntentFilter;
import com.rarnu.tools.neo.utils.PackageParserUtils;

import java.util.ArrayList;
import java.util.List;

public class CompInfo {

    /* PackageParser.Component<?> */
    public PackageParserUtils.Component component = null;
    public boolean enabled = false;
    public String fullPackageName = null;

    public String getCompName() {
        return component.className.substring(component.className.lastIndexOf(".") + 1);
    }

    public List<String> getIntents() {
        List<String> result = new ArrayList<>();
        if (component != null && component.intents != null) {
            for (IntentFilter a : component.intents) {
                if (a.countActions() > 0) {
                    for (int i = 0 ; i < a.countActions(); i ++) {
                        result.add(a.getAction(i));
                    }
                }
            }
        }
        return result;
    }

}

File: RootTools2/src/main/java/com/rarnu/tools/neo/data/CompInfo.kt
package com.rarnu.tools.neo.data

import com.rarnu.tools.neo.utils.PackageParserUtils

class CompInfo {

    /* PackageParser.Component<?> */
    var component: PackageParserUtils.Component? = null
    var enabled = false
    var fullPackageName: String? = null

    val compName: String?
        get() = component?.className?.substring(component?.className!!.lastIndexOf(".") + 1)

    val intents: List<String>
        get() {
            val result = arrayListOf<String>()
            if (component != null && component?.intents != null) {
                for (a in component!!.intents!!) {
                    if (a.countActions() > 0) {
                        for (i in 0..a.countActions() - 1) {
                            result.add(a.getAction(i))
                        }
                    }
                }
            }
            return result
        }

}
